Update dev-MC_25km_jra_ryf

dougiesquire opened the pull request:
<!-- use these prompts for changes to configuration branhces, skip them for main branch changes -->
**1. Summary**:

This PR includes updates for the upcoming 1.0-beta release discussed in our meeting this morning:
- Set `ENTHALPY_FROM_COUPLER = False` in `MOM_input` - see #631 
- Set `MAX_P_SURF = -1.0` in `MOM_input` - see #634 
- Use `ROQUET_RHO` EOS - see #235
- Remove diagnostics on rho2 coordinates by default.
- I also address some review comments from https://github.com/ACCESS-NRI/access-om3-configs/pull/622#discussion_r2196305107 that @aekiss posted after that PR was merged

**2. Issues Addressed:**
<!-- Add links to github issue(s) this is related to -->
- See above

**3. Depedencies (e.g. on payu, model or om3-scripts)**

This change requires changes to (note required version where true):
- [ ] payu:
- [ ] access-om3:
- [ ] om3-scripts:
<!-- Describe and link to the related changes to dependencies -->

**4. Ad-hoc Testing**

This configuration has been run for 20 years

**5. CI Testing**
<!-- Has the CI-testing been run? -->
- [ ] `!test repro` has been run

**6. Reproducibility**

Is this reproducible with the previous commit? (If not, why not?)
- [ ] Yes
- [ ] No - `!test repro commit` has been run. <!-- add detail below for why it's answer changing --> 

**7. Documentation**
<!--Does this impact documentation? Has the wiki been updated? Have the `docs/MOM_*` files been updated ?-->

The docs folder has been updated with output from running the model?
- [x] Yes
- [ ] N/A

A PR has been created for updating the documentation?
- [ ] Yes: <!--link-->
- [ ] N/A

**8. Formatting**
<!-- Are changes to MOM_input in the same order as docs/MOM_parameter_docs.short? -->

Changes to MOM_input have been copied from model output in docs/MOM_parameter_docs.short?
- [x] Yes
- [ ] N/A

**9. Merge Strategy**
<!-- What is the planned merge strategy (Merge commit, Rebase and merge, or squash) ?
If not squash, link to the related issue in the commit descriptions -->

- [ ] Merge commit
- [x] Rebase and merge
- [ ] Squash

minghangli-uni commented on diagnostic_profiles/diag_table_standard:
```diff
@@ -67,15 +67,6 @@ ACCESS-OM3
-"access-om3.mom6.3d.e.rho2.1mon.mean.%4yr", 1, "months", 1, "days", "time", 1, "years"
-"ocean_model_rho2", "e", "e", "access-om3.mom6.3d.e.rho2.1mon.mean.%4yr", "all", "average", "none", 2
-
-"access-om3.mom6.3d.umo.rho2.1mon.mean.%4yr", 1, "months", 1, "days", "time", 1, "years"
-"ocean_model_rho2", "umo", "umo", "access-om3.mom6.3d.umo.rho2.1mon.mean.%4yr", "all", "average", "none", 2
-
-"access-om3.mom6.3d.vmo.rho2.1mon.mean.%4yr", 1, "months", 1, "days", "time", 1, "years"
-"ocean_model_rho2", "vmo", "vmo", "access-om3.mom6.3d.vmo.rho2.1mon.mean.%4yr", "all", "average", "none", 2
```
Is there a gh related issue for removing them?

minghangli-uni replied:
Just to confirm, are these two not needed, even when using z coord?

aekiss replied:
yep, I think so

aekiss commented on diagnostic_profiles/source_yaml_files/diag_table_standard_source.yaml:
```diff
@@ -87,7 +87,7 @@ global_defaults:
         'True': mean
         None: ""  # for empty items
-    z_coord: ""
+    file_name_diag_coord: "" # name of diagnostic vetrical coordinate as specified in DIAG_COORDS in MOM_input; use "" for native coordinate
```
```suggestion
    file_name_diag_coord: "" # name of diagnostic vertical coordinate as specified in DIAG_COORDS in MOM_input; use "" for native coordinate
```

aekiss replied:
LOL, I assumed it was me :-)

aekiss reviewed (dismissed):
LGTM, thanks @dougiesquire

dougiesquire commented:
I've made a couple of changes:
- I've added back in the diags on rho2 following discussion in https://github.com/ACCESS-NRI/access-om3-configs/issues/642
- I noticed that with `USE_CONTEMP_ABSSAL = True` the 3D temp and salt diagnostics are not available via their cmor names, so I've changed those to their MOM names `temp` and `salt`

@minghangli-uni / @anton-seaice could you please re-review

anton-seaice reviewed (approved):
Looks ok to me :)
